--- gmime-3.2.7/configure.ac.orig	2025-03-03 20:31:22.015850362 +0100
+++ gmime-3.2.7/configure.ac	2025-03-03 21:09:08.510238374 +0100
@@ -484,13 +484,13 @@ dnl ****************************
 dnl *** Checks for libsocket ***
 dnl ****************************
 LIBSOCKET=""
-AC_CHECK_LIB(socket, getaddrinfo, LIBSOCKET="-lsocket")
+AC_CHECK_FUNC(getaddrinfo,,[AC_CHECK_LIB(socket, getaddrinfo, LIBSOCKET="-lsocket")])
 
 dnl *************************
 dnl *** Checks for libnsl ***
 dnl *************************
 LIBNSL=""
-AC_CHECK_LIB(nsl, getaddrinfo, LIBNSL="-lnsl")
+AC_CHECK_FUNC(getaddrinfo,,[AC_CHECK_LIB(nsl, getaddrinfo, LIBNSL="-lnsl")])
 
 dnl **************************************************
 dnl *** Checks for GpgME needed for PGP and S/MIME ***
